nrf-hal-common/src/delay_timer.rs

```diff
@@ -0,0 +1,56 @@
+//! Implementations of the embedded-hal `Delay` trait using Timers
+
+use cast::u32;
+
+use embedded_hal::blocking::delay::{DelayMs, DelayUs};
+
+use crate::timer::{self, Timer};
+pub struct Delay<T, U>(Timer<T, U>);
+
+impl<T: timer::Instance, U> Delay<T, U> {
+    /// Configures a Timer as a delay provider
+    pub fn new(timer: Timer<T, U>) -> Self {
+        Self(timer)
+    }
```

Comment:
Would it be a problem to simply implement the `DelayMs` and `DelayUs` traits on `Timer<T, U>` directly?